src/twister2/device/hardware_adapter.py

```diff
@@ -17,16 +17,14 @@
 
 class HardwareAdapter(DeviceAbstract):
 
-    def __init__(self, twister_config, hardware_map: HardwareMap | None = None):
+    def __init__(self, twister_config, hardware_map: HardwareMap | None = None) -> None:
         if hardware_map is None:
             raise TwisterException('Hardware map must be provided for hardware adapter')
         super().__init__(twister_config, hardware_map=hardware_map)
-        self.board_id: str = self.hardware_map.probe_id or self.hardware_map.id
-        self.connection: Optional[serial.Serial] = None
-        self.timeout = 1  # serial timeout
+        self.connection: serial.Serial | None = None
 
-    def connect(self):
-        """Open connection."""
+    def connect(self) -> serial.Serial:
```

Comment:
This method should also have an argument `timeout`. Otherwise when you call `connect` before `flash`, then `Serial` object obtains always default value of `self.timeout` (60.0 sec).

Reply:
I took this implementation from the old twister, but I have a doubt what value should be used as timeout. From `serial` package documentation this timeout is usually 0 or 1 second (60 seconds seems too big). Maybe we should use default value like 1 seconds? We can change it in the future if it is required.

Reply:
We can return line only if it it is not empty string. 
```python
if line := self.connection.readline().decode('UTF-8').strip():
    yield line
```